### DISCUSSION TITLE
ulong support

### DISCUSSION THREAD

**Discussion by Unreal852** [realm/realm-dotnet#3445] [Q&A] [answered]
Hi guys, i got a small Discord bot currently using LiteDB for its database. I'm in the process of migrating to RealmDB but i found out that it doesn't support ULong64 (which is used by everywhere by discord) so i would like to know if Realm will support this. Thanks.

**Comment by nirinchev** [comment]
There are no immediate plans to support this, but since `long` and `ulong` have the same size, you could just map a long property to `ulong`. It's a bit of boilerplate, but all the complexity can be hidden within your model definitions like:

```csharp
partial class MyModel : IRealmObject
{
    [MapTo(nameof(Value))] // MapTo is useful for .Filter queries where you're writing a query string by hand
    private long ValueLong { get; set; }

    public ulong Value
    {
        get => (ulong)ValueLong;
        set => ValueLong = (long)value;
    }

    // This is needed if you're using change notifications to be notified that the remapped property has changed.
    partial void OnPropertyChanged(string? propertyName)
    {
        if (propertyName == nameof(ValueLong))
        {
            RaisePropertyChanged(nameof(Value));
        }
    }
}
```

**Reply by Unreal852** [reply]
There is something wrong, or maybe i am wrong. 

I created a quick console project and i wrote this
``` CSharp
using Realms;

ulong ulongId = 1062062378667233280;

var dbFile = Path.Combine(AppContext.BaseDirectory, "db.realm");
var realm = Realm.GetInstance(new RealmConfiguration(dbFile));
realm.Write(() => { realm.Add(new MyModel { MyId = (long)ulongId }); });

Console.ReadKey();

class MyModel : RealmObject
{
    public long MyId { get; set; }
}
```

So, in theory, the value inside the Realm should be the same as my `ulongId` so in this case `1062062378667233280` but it's not, see the picture below.
![image](https://github.com/realm/realm-dotnet/assets/5130243/9a65a3e0-dfd0-4992-8694-3d808ca77fe2)

**Reply by Unreal852** [reply]
I just tried to read the database from code, and it returns the right value. It might be something related to Realm Studio and not Realm.NET. Sorry for the inconvenience.

**Reply by fealebenpae** [reply, marked as answer]
Realm doesn’t natively support unsigned integers, so Realm Studio interprets the field as a signed integer.
You could use the `decimal` type for your field which is big enough to contain an UInt64 without having to reinterpret it and should visualize correctly in Studio, I believe.

**Reply by Unreal852** [reply]
Yep, i just tried and you're right. Thanks
